Sources/KSCrashRecordingCore/KSThread.c

```diff
@@ -52,6 +68,34 @@ bool ksthread_getThreadName(const KSThread thread, char *const buffer, int bufLe
     return pthread_getname_np(pthread, buffer, (unsigned)bufLength) == 0;
 }
 
+int ksthread_getThreadState(const KSThread thread)
```

Comment:
You could simplify this a bunch to the following:

```
static int ksthread_getThreadState(const KSThread thread)
{
    thread_basic_info_t info = {0};
    mach_msg_type_number_t count = THREAD_BASIC_INFO_COUNT;
    kern_return_t kr = 0;
    
    kr = thread_info((thread_t)thread, THREAD_BASIC_INFO, (thread_info_t)info, &count);
    return  kr == KERN_SUCCESS ? info->run_state : TH_STATE_UNSET;
}
```

Reply:
> memory safe function end up calling into vm_read_overwrite

That’s a good point. What solutions are there?

Reply:
I’m not sure. What’s the validation for in these cases anyway? 

I don’t think we need to validate memory returned by these system functions. I imagine those men validation functions were only meant to be used in the memory monitor.